arrangor/src/main/kotlin/no/nav/amt/tiltak/veileder/ArrangorVeilederServiceImpl.kt

```diff
@@ -0,0 +1,113 @@
+package no.nav.amt.tiltak.veileder
+
+import no.nav.amt.tiltak.core.domain.arrangor.ArrangorVeileder
+import no.nav.amt.tiltak.core.domain.arrangor.ArrangorVeilederInput
+import no.nav.amt.tiltak.core.domain.tilgangskontroll.ArrangorAnsattRolle
+import no.nav.amt.tiltak.core.exceptions.ValidationException
+import no.nav.amt.tiltak.core.port.ArrangorAnsattTilgangService
+import no.nav.amt.tiltak.core.port.ArrangorVeilederService
+import no.nav.amt.tiltak.core.port.DeltakerService
+import no.nav.amt.tiltak.core.port.GjennomforingService
+import org.springframework.stereotype.Service
+import java.time.ZonedDateTime
+import java.util.*
+
+@Service
+class ArrangorVeilederServiceImpl (
+	private val arrangorVeilederRepository: ArrangorVeilederRepository,
+	private val arrangorAnsattTilgangService: ArrangorAnsattTilgangService,
+	private val deltakerService: DeltakerService,
+	private val gjennomforingService: GjennomforingService,
+): ArrangorVeilederService {
+
+	private val maksMedveiledere = 3
+
+	override fun opprettVeiledere(veilederInputs: List<ArrangorVeilederInput>, deltakerIder: List<UUID>) {
+		if (veilederInputs.filter { !it.erMedveileder }.size > 1) {
+			throw ValidationException("Deltakere kan kun ha 1 veileder.")
+		}
+
+		verifiserVeilederTilganger(deltakerIder, veilederInputs.map { it.ansattId })
+
+		val veiledere = veilederInputs.map { OpprettVeilederDbo(it.ansattId, it.erMedveileder) }
+
+		inaktiverVeiledereSomSkalErstattes(veiledere, deltakerIder)
+
+		arrangorVeilederRepository.opprettVeiledere(veiledere, deltakerIder)
+	}
+
+	override fun hentVeiledereForDeltaker(deltakerId: UUID): List<ArrangorVeileder> {
+		return arrangorVeilederRepository.getAktiveForDeltaker(deltakerId).map { it.toArrangorVeileder() }
+	}
+
+	private fun inaktiverVeiledereSomSkalErstattes(veiledere: List<OpprettVeilederDbo>, deltakerIder: List<UUID>) {
+		val antallNyeMedveiledere = veiledere.count { it.erMedveileder }
+
+		if (antallNyeMedveiledere > maksMedveiledere)
+			throw ValidationException("Kan ikke håndtere flere enn $maksMedveiledere medveiledere")
+
+		arrangorVeilederRepository.inaktiverVeiledereForDeltakere(
+			veiledere.map { it.ansattId },
+			deltakerIder,
+		)
+
+		val aktiveVeiledere = arrangorVeilederRepository.getAktiveForDeltakere(deltakerIder)
+
+		val veiledereSomSkalErstattes = aktiveVeiledereSomSkalErstattes(veiledere, aktiveVeiledere)
+			.plus(aktiveMedveiledereSomSkalErstattes(aktiveVeiledere, antallNyeMedveiledere))
+
+		arrangorVeilederRepository.inaktiverVeiledere(veiledereSomSkalErstattes)
+	}
+
+	private fun aktiveMedveiledereSomSkalErstattes(
```

Comment:
Inneholder input alle medveilederne man ønsker å beholde? Da kan du i så fall sende inn listen med veiledere fra input og listen med eksisterende medveiledere (fra databasen) og så returnert alle id-ene som finnes i listen over eksisterende veiledere, men ikke i listen over veiledere fra input (altså de som ikke lenger skal brukes). 

Hvis inputen inneholder de medveilederne man ønsker å sitte igjen med etter oppdatering så kan du sjekke tidlig at antallet ikke er for mange også :)